src/core/fsqual.cc

```diff
@@ -98,6 +98,7 @@ bool filesystem_has_good_aio_support(sstring directory, bool verbose) {
         std::cout << "context switch per appending io: " << rate
                   << " (" << verdict << ")\n";
     }
+    free(buf);
```

Comment:
nit, might want to use 
```c++
auto del = defer([&] () noexcept { ::free(buf); });
```
right after the `aligned_alloc()` call for better readability?

Reply:
I concur on this nit. After allocation there are some exception throws, so the buffer would still leak. @syuu1228 , please

Reply:
> nit, might want to use
> 
> ```c++
> auto del = defer([&] () noexcept { ::free(buf); });
> ```
> 
> right after the `aligned_alloc()` call for better readability?

Replaced `free(buf)` with `auto del = defer([&] () noexcept { ::free(buf); })`.